src/core/toga/widgets/switch.py

```diff
@@ -9,85 +11,156 @@ class Switch(Widget):
         and False (off, unchecked).
 
     Args:
-        label (str): Text to be shown next to the switch.
+        text (str): Text to be shown next to the switch.
         id (str): AN identifier for this widget.
         style (:obj:`Style`): An optional style object.
             If no style is provided then a new one will be created for the widget.
-        on_toggle (``callable``): Function to execute when pressed.
-        is_on (bool): Current on or off state of the switch.
+        on_change (``callable``): Function to execute when pressed.
+        value (bool): Current on or off state of the switch.
         enabled (bool): Whether or not interaction with the button is possible, defaults to `True`.
         factory (:obj:`module`): A python module that is capable to return a
             implementation of this class with the same name. (optional & normally not needed)
     """
 
     def __init__(
             self,
-            label,
+            text,
             id=None,
             style=None,
-            on_toggle=None,
-            is_on=False,
+            on_change=None,
+            on_toggle=None,  # DEPRECATED!
+            is_on=None,
+            value=False,
             enabled=True,
             factory=None,
     ):
         super().__init__(id=id, style=style, factory=factory)
 
         self._impl = self.factory.Switch(interface=self)
 
-        self.label = label
-        self.on_toggle = on_toggle
-        self.is_on = is_on
+        self.text = text
+        if on_toggle:
+            self.on_toggle = on_toggle
```

Comment:
This approach is good; but we should also include:
1. A code comment describing that this is a backward compatibility shim
2. Error handling for the case where someone sets *both* `on_toggle` and `on_change`. 

Even better would be to make the "backwards compatibility" portion of the code as simple as possible, leaning into the "future state" - that is, make the backwards compatibility code something that can be cleanly deleted, rather than something where the logic will need to be re-written. 
```
# 2022-07: Backwards compatibility: on_toggle replace with on_change
if on_toggle:
    if on_change:
        raise error
    else:
        on_change = on_toggle
# end backwards compatibility.

self.on_change = on_change
```
That way, the future code change is a clean deletion of 7 lines of code, rather than a complex revision.